jsettlers.logic/src/jsettlers/logic/map/grid/MainGrid.java

```diff
@@ -853,12 +885,16 @@ public final short getHeight() {
 			return height;
 		}
 
-		boolean canConstructAt(short x, short y, EBuildingType type, byte playerId) {
+		@Override
+		public boolean canConstructAt(short x, short y, EBuildingType type, byte playerId) {
 			ELandscapeType[] landscapes = type.getGroundtypes();
 			RelativePoint[] protectedTiles = type.getProtectedTiles();
 
 			int firstPosX = protectedTiles[0].calculateX(x);
 			int firstPosY = protectedTiles[0].calculateY(y);
+			if (firstPosX < 0 || firstPosY < 0 || firstPosX > partitionsGrid.getWidth() || firstPosY > partitionsGrid.getHeight()) {
```

Comment:
We have an inMap()-Method for this somewhere ;-).

And we should check this for the whole building, not just the start position.